crates/rpc/ipc/src/server/mod.rs

```diff
@@ -139,7 +140,15 @@ where
             .to_fs_name::<GenericFilePath>()
             .and_then(|name| ListenerOptions::new().name(name).create_tokio())
         {
-            Ok(listener) => listener,
+            Ok(listener) => {
+                if let Some(perms_str) = &self.cfg.ipc_socket_permissions {
+                    if let Ok(mode) = u32::from_str_radix(&perms_str.replace("0o", ""), 8) {
```

Comment:
this should do it, can't think of a better way to do this, and I also don't think we should parse this on cli because configuring string seems easier for this